### ISSUE TITLE
Wagtail userbar should label its complementary landmark

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#14184]
### Issue summary

Spotted by @helenb - our userbar template uses an `<aside>` element without giving it a label: https://github.com/wagtail/wagtail/blob/e618bf21fac33f555a3b932b5a0a75e09e626e59/wagtail/admin/templates/wagtailadmin/userbar/base.html#L4-L5

We need to label it so that if there are multiple complementary landmarks on the page, each is separately identifiable. This can most likely be done with a `aria-labelledby` pointing to the userbar toggle button:

```html
<aside {% if request.in_preview_panel %}hidden{% endif %} aria-labelledby="wagtail-userbar-trigger">
```

### Steps to reproduce

1. Go to [this static homepage demo](https://static-wagtail-v7-3.netlify.app/) or any other page with the userbar present
2. Inspect the page landmarks and note the aside/complementary that has no label

### Can be reproduced

Yes, on the bakerydemo

### Technical details

N/A

### Working on this

View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you're ready to start.

Adding the extra attribute should be very simple, but it’s important to test this properly via browser developer tools or assistive technologies to confirm this works. And maybe a unit test in the userbar code for the presence of `aria-labelledby`?

**Comment by Kkt04** [comment]
Can i work on this?

**Comment by laymonage** [comment]
Yes

**Comment by zanzo2003** [comment]
@laymonage @thibaudcolas  is this already assigned to someone ? if not I would like to try this.
